### ISSUE TITLE
Does 'llm-distillation/datasets/generator.pygenerator.py' need to generate new dataset before load_from_disk?

### ISSUE THREAD

**Issue by haiduo** [Nicolas-BZRD/llm-recipes#1]
Hi, thanks for the open source, great work！

When I use load_from_disk (squad data) to load "Nicolas-BZRD/uld_loss_Llama-2-7b-chat-hf-qed", it shows that 7B-squad is neither a `Dataset` directory nor a `DatasetDict` directory appears. Do you think it's necessary first? Processed by "llm-distillation/datasets/generator.py" before it can be used? If yes, how do you set the parameters? 
`https://github.com/Nicolas-BZRD/llm-distillation/blob/bba10992fc60c0e0158fceb7438bc5e25bfc7a1a/datasets/generator.py#L35C4-L48C113

Looking forward to your reply！

**Comment by haiduo** [comment]
I find that changing this line:  
https://github.com/Nicolas-BZRD/llm-distillation/blob/bba10992fc60c0e0158fceb7438bc5e25bfc7a1a/datasets/loader/squad.py#L70  
  to: `dataset = load_dataset("Nicolas-BZRD/uld_loss_Llama-2-7b-chat-hf-squad")` do the trick.

I have also thoroughly studied your data processing process. Generally, the raw data is first processed by generator.py, then split into train and val by process.py, and finally saved locally for subsequent training processing.

In addition, I have three questions that I still need your answers to.
1. I feel like the calculation of Wasserstein distance is too simple.
```
size = min(student_answer_size[i], teacher_answer_size[i])
distillation_loss[i] = abs(student[i][:size] - teacher[i][:size]).sum(-1).mean(-1)
```
Although the effect is proven in your paper, I think it needs to be verified on a larger data set to be more convincing, such as ['EleutherAI/lm-evaluation-harness'](https://github.com/EleutherAI/lm-evaluation-harness) or ['cerebras/SlimPajama-627B'](https://huggingface.co/datasets/cerebras/SlimPajama-627B)

2. Why is it necessary to skip generating token=eos for the student model, but not for the teacher model?
```
Skip student eos: True
Skip teacher eos: False
```

3. When uniform cost is used, that is, the optimal transmission cost T=1, then why is the padding operation used for vocabulary size instead of directly clipping to the smallest dimension to get the same vocabulary size? Or have you tried any experimental comparison of the clipping to the smallest dimension? Because I may think that the error in calculating the Wasserstein distance (the absolute value of the difference) will be greater after the padding than clipping.  
```
diff_size = student.size(2) - teacher.size(2)
if diff_size > 0:
   teacher = F.pad(teacher, (0, diff_size), value=0)
elif diff_size < 0:
   student = F.pad(student, (0, abs(diff_size)), value=0)
```

**Comment by Nicolas-BZRD** [comment]
Hi Haiduo,

Thank you for your interest on the paper. Indeed, it's crucial to adjust the path to a HuggingFace ID if you aim to load it directly from the Hub. In our scenario, the node where computations were conducted lacked internet connectivity, necessitating pre downloading the dataset.

Regarding your queries:

1 - Why do you find it too simple.

It could be beneficial to test it on larger datasets, but this would escalate computational costs. Consequently, we opt to concentrate our experimentation on numerous tasks rather than a single large dataset.

2 - We still generate the eos_token for the student but didn't compute the loss on it. It's because the teacher model (for instruct model) didn't always generate an eos_token at the end of this answer so we can't compute the loss for it.

3 - We opt to pad to ensure that the sum of the logits equals 1. This approach ensures that the sum of the logits for both teacher and student is consistent (1).

**Comment by haiduo** [comment]
Hi Nicolas-BZRD,

Thank you for your prompt reply and solving my problem.🙂
